Doc/source/designspaceLib/xml.rst

```diff
@@ -53,6 +53,29 @@ Overview
             </dict>
         </lib>
     </designspace>
+    
+=====
+Types
+=====
+
+- **number**: A fixed point value `equivalent to Fixed in OpenType <https://docs.microsoft.com/en-us/typography/opentype/spec/otff#data-types>`_ (a 32-bit signed fixed-point number (16.16))
+- **font design (unsigned) float**: A floating point value `equivalent to (U)FWORD in OpenType <https://docs.microsoft.com/en-us/typography/opentype/spec/otff#data-types>`_
+- **float**: A floating point value `equivalent to F2DOT14 in OpenType <https://docs.microsoft.com/en-us/typography/opentype/spec/otff#data-types>`_
+
+     The F2DOT14 format consists of a signed, 2's complement integer and an unsigned fraction.
+
+- **(unsigned) integer**
+- **boolean**: A true/false value, may be expressed as ``0`` or ``1``
```

Comment:
I think you're uncovering a real consistency issue here with booleans because some boolean fields are written as 0/1, e.g. axis hidden=1

https://github.com/fonttools/fonttools/blob/42e4d66184be1acda05bed1896da6d61c084835e/Lib/fontTools/designspaceLib/__init__.py#L1454-L1455

and then hidden is read back as "any provided value means True" apparently:

https://github.com/fonttools/fonttools/blob/42e4d66184be1acda05bed1896da6d61c084835e/Lib/fontTools/designspaceLib/__init__.py#L1861-L1862

and some others are read/written as true/false, e.g. elidable and oldersibling

https://github.com/fonttools/fonttools/blob/42e4d66184be1acda05bed1896da6d61c084835e/Lib/fontTools/designspaceLib/__init__.py#L1485-L1488

https://github.com/fonttools/fonttools/blob/42e4d66184be1acda05bed1896da6d61c084835e/Lib/fontTools/designspaceLib/__init__.py#L1902-L1903

So in the end I would say that in addition to making the documentation about booleans more uniform, the PR should also edit the code to make the real situation also uniform. Maybe something like:
* when writing always output 0/1
  * because that's how it's been done so far (the 2 true/false examples above are recent additions from Nikolaus and me, I think they're an oversight because we haven't paid enough attention to the existing convention; I would say it's OK since these two new fields are still young to change the default writer to write 1 instead of true)
* when  reading accept any of "0", "1", "true", "false" and interpret it correctly
  * in the spirit of TCP wisdom "be conservative in what you do, be liberal in what you accept from others", and because I think someone reading the amended documentation quickly, seeing "boolean" as a type and not reading the introduction about types where it's mentioned that the format uses 0/1, would naturally try to write true/false.
  
What do you think?

Reply:
Yep, that's a real problem all right. I'll add code changes as well.